src/Pos/Ssc/Extra/Logic.hs

```diff
@@ -181,34 +183,16 @@ type SscGlobalVerifyMode ssc m =
     (MonadSscMem ssc m, SscGStateClass ssc, WithLogger m,
      MonadDB m, MonadError (SscVerifyError ssc) m, WithNodeContext ssc m)
 
-sscRunGlobalUpdatePure
-    :: forall ssc a.
-       LoggerName
-    -> SscGlobalState ssc
-    -> LoggerNameBox (PureLogger (State (SscGlobalState ssc))) a
-    -> (a, SscGlobalState ssc, [LogEvent])
-sscRunGlobalUpdatePure loggerName st =
-    convertRes . flip runState st . runPureLog . usingLoggerName loggerName
-  where
-    convertRes :: ((a, [LogEvent]), SscGlobalState ssc)
-               -> (a, SscGlobalState ssc, [LogEvent])
-    convertRes ((res, events), newSt) = (res, newSt, events)
-
 sscRunGlobalUpdate
     :: forall ssc m a.
        SscGlobalApplyMode ssc m
-    => LoggerNameBox (PureLogger (State (SscGlobalState ssc))) a -> m a
+    => NamedPureLogger (State (SscGlobalState ssc)) a -> m a
 sscRunGlobalUpdate action = do
-    loggerName <- getLoggerName
     globalVar <- sscGlobal <$> askSscMem
-    (res, events) <- atomically $ sscRunGlobalUpdateDo loggerName globalVar
-    res <$ dispatchEvents events
+    launchNamedPureLog (atomically . evalStateTUninitialized) $
```

Comment:
I don't quite understand how it works... `launchNamedPureLog` is a functions, that takes named logging monadic actions and returns another action. How do you able to pass two monadic arguments to it?